src/QsCompiler/Tests.Compiler/TestCases/QirEntryPointTests/TestUnitReceivesBool.cpp

```diff
@@ -0,0 +1,11 @@
+﻿//------------------------------------------------------------------------------
+// This code was generated by a tool.
+// <auto-generated />
+//------------------------------------------------------------------------------
+
+extern "C" void UseBoolArg(); // NOLINT
+
+int main(int argc, char* argv[])
+{
+    return 0;
+}
```

Comment:
1. Would be nice to have some more comments (in the code or in the PR) about why we need an auto-generated file in the repo.
2. Rather curious that a function is declared `UseBoolArg()` that is not called by anyone.

Reply:
Regarding the first comment. This is a test case file that represents the expected generated C++ code. It is read by the test at _src/QsCompiler/Tests.Compiler/EntryPointGenerationTests.fs_ which compares it to the actual generated C++ code using _src/QsCompiler/Tests.Compiler/TestUnitReceivesBool.json_ as input.

Regarding the second comment, this PR is just intends to setup a base to start the implementation of the C++ entry-point driver generator. Next week, Sam will be picking this up to complete the implementation. I'll add a comment to the generated code to make this clear for now.

Reply:
Added comment clarifying the current state of C++ code generation.